src/as-utils.c

```diff
@@ -595,64 +595,44 @@ as_ptr_array_to_strv (GPtrArray *array)
 /**
  * as_gstring_replace:
  * @string: The #GString to operate on
- * @search: The text to search for
+ * @find: The text to search for
  * @replace: The text to use for substitutions
  *
  * Performs multiple search and replace operations on the given string.
  *
  * Returns: the number of replacements done, or 0 if @search is not found.
  **/
 guint
-as_gstring_replace (GString *string, const gchar *search, const gchar *replace)
-{
-	gchar *tmp;
-	guint count = 0;
-	gsize search_idx = 0;
-	gsize replace_len;
-	gsize search_len;
+as_gstring_replace (GString *string, const gchar *find, const gchar *replace)
+{
+	/* note: This is a direct copy from GLib upstream (with whitespace
+	 * fixed spaces to tabs). Once we can depend on GLib 2.68, this copy
+	 * should be dropped and g_string_replace() used instead.
+	 *
+	 * GLib is licensed under the LGPL-2.1+.
```

Comment:
Can you throw the whole code of this function behind a GLib version check macro, so we fall back to the GLib-internal version as soon as it is available with newer GLib? That also makes the code self-explanatory and is easy to grep for once we do raise the minimum GLib version.
See `as_date_time_format_iso8601` for one such example: https://github.com/ximion/appstream/blob/master/src/as-utils.c#L1744-L1782 (there's quite a few more in AppStream's code)